core/src/main/scala/com/yahoo/maha/core/query/oracle/OracleQueryGenerator.scala

```diff
@@ -234,15 +234,19 @@ b. Dim Driven
       val dimSelect = dimSelectSet.mkString(", ")
       val dimWhere = generateWhereClause(dimBundle, subqueryBundles)
 
+      val isTotalRowRequest = isDimOnly && requestModel.isDimDriven && requestModel.maxRows == 1 && requestModel.includeRowCount
```

Comment:
This looks like special casing, the query generator should not know if it is a total rows request (e.g. requestModel.maxRows == 1).  The request model should construct correctly and query pipeline should execute correctly, the generator should not care.  Did you take a look at what Pranav suggested in your previous PR?

Reply:
I'm moving the change to QueryPipeline.  The SyncSingleEngine case is the only currently affected, but I am testing if I can insert the logic straight into the queryPipeline builder without harming other queries.